hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -0,0 +1,381 @@
+      if (table == null) {
+        LOG.error("Table " + tableName + " not found in OM Metadata.");
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+
+      try {
+        if (getTablesToCalculateSize().contains(tableName)) {
+          Triple<Long, Long, Long> details = getTableSizeAndCount(table);
+          objectCountMap.put(
+              getTableCountKeyFromTable(tableName), details.getLeft());
+          unReplicatedSizeCountMap.put(
+              getUnReplicatedSizeKeyFromTable(tableName), details.getMiddle());
+          replicatedSizeCountMap.put(
+              getReplicatedSizeKeyFromTable(tableName), details.getRight());
+        } else {
+          long count = getCount(table.iterator());
+          objectCountMap.put(getTableCountKeyFromTable(tableName), count);
+        }
+      } catch (IOException ioEx) {
+        LOG.error("Unable to populate Table Count in Recon DB.", ioEx);
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+    }
+
+    // Write count data to DB if the sizeCountMap is not empty
+        if (!objectCountMap.isEmpty()) {
+          writeDataToDB(objectCountMap);
+        }
+
+    // Write unreplicated size data to DB if the sizeCountMap is not empty
+        if (!unReplicatedSizeCountMap.isEmpty()) {
+          writeDataToDB(unReplicatedSizeCountMap);
+        }
+
+    // Write replicated size data to DB if the sizeCountMap is not empty
+        if (!replicatedSizeCountMap.isEmpty()) {
+          writeDataToDB(replicatedSizeCountMap);
+        }
+
+    LOG.info("Completed a 'reprocess' run of OmTableInsightTask.");
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+  /**
+   * Returns a pair with the total count of records (left), total unReplicated
+   * size (middle), and total replicated size (right) in the given table.
+   * Increments count for each record and adds the dataSize if a record's value
+   * is an instance of OmKeyInfo. If the table is null, returns (0, 0, 0).
+   *
+   * @param table The table from which to get the count and data size.
+   * @return Three Long values representing the count, unReplicated size,
+   *         and replicated size.
+   * @throws IOException If an I/O error occurs during the table iteration.
+   */
+  private Triple<Long, Long, Long> getTableSizeAndCount(Table table)
+      throws IOException {
+    long count = 0;
+    long unReplicatedSize = 0;
+    long replicatedSize = 0;
+
+    TableIterator<String, ? extends Table.KeyValue<String, ?>> iterator =
+        table.iterator();
+    while (iterator.hasNext()) {
+      Table.KeyValue<String, ?> kv = iterator.next();
+
+      if (kv != null && kv.getValue() != null) {
+        if (kv.getValue() instanceof OmKeyInfo) {
+          OmKeyInfo omKeyInfo = (OmKeyInfo) kv.getValue();
+          unReplicatedSize += omKeyInfo.getDataSize();
+          replicatedSize += omKeyInfo.getReplicatedSize();
+        }
+      }
+      count++;  // Increment count for each row
```

Comment:
The reason for checking if the kv value is an OmKeyInfo object is to ensure accurate calculation of the size for entries in the openKeyTable and openFileTable that store OmKeyInfo objects. By checking the type, we can retrieve the specific data size and replicated size from the OmKeyInfo object and update the corresponding variables (unReplicatedSize and replicatedSize) accordingly.

Regarding the count, there is no need to specifically to do a count++ for only OmKeyInfo instances as this method `getTableSizeAndCount` can be used by other tables as well for example `deletedTable` which contains RepeatedOmKeyInfo objects. 
```
      if (kv != null && kv.getValue() != null) {
        if (kv.getValue() instanceof OmKeyInfo) {
          OmKeyInfo omKeyInfo = (OmKeyInfo) kv.getValue();
          unReplicatedSize += omKeyInfo.getDataSize();
          replicatedSize += omKeyInfo.getReplicatedSize();
        }
      }
```

Reply:
You are already filtering using `getTablesToCalculateSize` to determine for which table we need to get size and count and calling getTableSizeAndCount in `reprocess` method. For the tables which need just count you are calling `getCount(table.iterator());` already. So in case of deletedTable do we need to call `getTablesToCalculateSize` which will just fill count ? Which actually can be done in getCount method.

Reply:
Thanks, @ashishkumar50, for your comment!

I fully agree that in our current code, we have not included the deletedTable in the list of tables returned by `getTablesToCalculateSize`. This is because the current pull request specifically focuses on `openKeys`. However, we do have a separate PR that addresses pending deleted keys in the `deletedTable`, and we need to calculate their total sizes as well. This work is being tracked by the following JIRA ticket: https://issues.apache.org/jira/browse/HDDS-8586.

Once this patch is merged, we will add the deletedTable in the list of tables to calculate the size. Additionally, we will need to introduce another check similar to the one mentioned above, which verifies if the value returned by the table is of type RepeatedOmKeyInfo or not. This check will help handle the size calculations for the pending deleted keys.